pkg/cloudprovider/aws/fleet/kubeprovider.go

```diff
@@ -0,0 +1,31 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fleet
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/client-go/kubernetes"
+)
+
+type KubeProvider struct {
+	clientSet *kubernetes.Clientset
+}
+
+func (k *KubeProvider) Version() (string, error) {
```

Comment:
I am conflicted between which package should this file be in, `cloudprovider` or `allocator`. Open to suggestions.

Reply:
I might skip this package entirely. The formatting you're doing is specific to the AMI, so I'd write that logic there. It's pretty slick that you can just call ` k.clientSet.Discovery().ServerVersion()` wherever you need the version.